### ISSUE TITLE
📎 Implement `useImmediateReturn`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#4748]
### Description

Port the [`prefer-immediate-return`](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/prefer-immediate-return.md) rule from `eslint-plugin-sonarjs`.

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by ematipico** [comment]
This sounds like more an assist action rather than a lint rule 🤔

**Comment by florianmatz** [comment]
@ematipico eslint itself has kind of the same rule - [arrow-body-style](https://eslint.org/docs/latest/rules/arrow-body-style) when configured to ‘as-needed’ or ‘never’.

**Comment by ematipico** [comment]
Yeah but this is a styling thing and it doesn't fit our rule pillars. An assist action seems to cover that

**Comment by florianmatz** [comment]
🤷 - this can be said for a bunch of rules like ‘ useConsistentCurlyBraces’? Also more style related IMHO and still useful / relevant? I don’t mean to be overly clever but if every rule that is even available in the core of eslint is negated where does that leave biome / the community?

**Comment by ematipico** [comment]
Yes, there's a good chance we're going to move it

**Comment by dyc3** [comment]
After looking at it again, I agree that this seems like an assist. Do we have guidelines (like our rule pillars) for what should be a lint vs assist?

**Comment by ematipico** [comment]
@dyc3 

I usually use our [rule pillars](https://biomejs.dev/linter/#rule-pillars) to understand if a rule should be part of the linter or not.

Another thing that I use as a guideline is editors. IntelliJ comes to mind, for example. For `useImmediateReturn`, this is usually suggested as a refactor opportunity by the editor.

**Comment by marc-at-brightnight** [comment]
If the proposal on this issue is to implement eslint's arrow-body-style as-needed, I would +1 that. Basically, I want a rule to disallow:
```
const fn = () => {
  return 5;
};
```
in favor of
```
const fn = () => 5;
```

I'm tired of commenting on this in people's PRs.

**Comment by dyc3** [comment]
@marc-at-brightnight This task is for the rule linked in the issue body. `arrow-body-style` is a different rule.

**Comment by ematipico** [comment]
No, it's not. It's to remove the intermediate variable

**Comment by marc-at-brightnight** [comment]
> No, it's not. It's to remove the intermediate variable

should I open a different issue on this? (:

**Comment by ematipico** [comment]
No, it's not. It's to remove the intermediate variable

**Comment by dyc3** [comment]
Sure, go ahead.
